akka-http-core/src/main/scala/akka/http/engine/client/HostConnectionPoolConductor.scala

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.client
+
+import language.existentials
+import scala.annotation.tailrec
+import scala.collection.immutable
+import akka.event.LoggingAdapter
+import akka.stream.scaladsl._
+import akka.stream._
+import akka.http.model.HttpMethod
+import akka.http.util._
+
+private object HostConnectionPoolConductor {
+  import HostConnectionPoolGateway.RequestContext
+  import HostConnectionPoolSlot.{ SlotEvent, SimpleSlotEvent }
+
+  case class Ports(
+    requestIn: Inlet[RequestContext],
+    slotEventIn: Inlet[SlotEvent],
+    slotOuts: immutable.Seq[Outlet[RequestContext]]) extends Shape {
+
+    override val inlets = requestIn :: slotEventIn :: Nil
+    override def outlets = slotOuts
+
+    override def deepCopy(): Shape =
+      Ports(
+        new Inlet(requestIn.toString),
+        new Inlet(slotEventIn.toString),
+        slotOuts.map(o ⇒ new Outlet(o.toString)))
+
+    override def copyFromPorts(inlets: immutable.Seq[Inlet[_]], outlets: immutable.Seq[Outlet[_]]): Shape =
+      Ports(
+        inlets.head.asInstanceOf[Inlet[RequestContext]],
+        inlets.last.asInstanceOf[Inlet[SlotEvent]],
+        outlets.asInstanceOf[immutable.Seq[Outlet[RequestContext]]])
+  }
+
+  /*
+    Stream Setup
+    ============
+                                                                                                  Request- 
+    Request-   +-----------+     +-----------+    Switch-    +-------------+     +-----------+    Context
+    Context    |   merge1  |     |   merge2  |    Command    |   doubler   |     |   route   +-------------->
+    +--------->|  (Merge-  +---->|  (Flexi   +-------------->| (MapConcat) +---->|  (Flexi   +-------------->
+               | Preferred)|     |   Merge)  |               |             |     |   Route)  +-------------->
+               +----+------+     +-----+-----+               +-------------+     +-----------+       to slots     
+                    ^                  ^                                                               
+                    |                  | SimpleSlotEvent                                              
+                    | Request-         |                                                               
+                    | Context     +---------+
+                    +-------------+  retry  |<-------- Slot Event (from slotEventMerge)
+                                  |  Split  |
+                                  +---------+
+
+  */
+  def apply(slotCount: Int, maxRetries: Int, pipeliningLimit: Int, log: LoggingAdapter): Graph[Ports, Any] =
+    FlowGraph.partial() { implicit b ⇒
+      import FlowGraph.Implicits._
+
+      val merge1 = b.add(new Merge1)
+      val merge2 = b.add(new Merge2(slotCount, maxRetries, pipeliningLimit, log))
+      val doubler = Flow[SwitchCommand].mapConcat(x ⇒ x :: x :: Nil)
```

Comment:
It's a work-around for https://github.com/akka/akka/issues/17004.
I'll add a respective comment.